### PR TITLE
Feature/core 849 digital twin general overview

### PR DESCRIPTION
**Pull request by datrinh**
Currently missing due to ongoing backend work:

- createdAt timestamp
- transaction timestamp

### CODE REVIEW

File: dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts [hunk 1]
```diff
@@ -118,9 +118,10 @@ export default class DigitalTwinDetailComponent extends mixins(EvanComponent) {
   }
 
   /**
-   * Cuts the description to a maximum of 300 characters.
+   * Limits the description to a certain maximum amount of characters.
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/DigitalTwinDetail.ts]:
```suggestion
   * Truncates the description to a certain amount of characters.
```

File: dapps/digital-twin-detail/src/components/overview/DetailOverview.ts [hunk 2]
```diff
@@ -17,13 +17,103 @@
...
+    }));
+  }
+
+  /**
+   * Enhance the current twin with createdAt timestamp
+   */
+  async attachCreatedAt(): Promise<void> {
+    const authHeaders = await utils.getSmartAgentAuthHeaders(this.runtime);
+    const core = this.runtime.environment === 'testcore' ? '.test' : '';
+    const url = `https://search${core}.evan.network/api/smart-agents/search`;
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/overview/DetailOverview.ts]: why not put all the request logic in a vue-agnostic service? Would also have the benefit of determining the request Url only once on initialization and the component would be way smaller.

Comment (datrinh) [reply dapps/digital-twin-detail/src/components/overview/DetailOverview.ts]: Done. authheaders can't be in constructor cuz of its async nature.

File: dapps/digital-twin-detail/src/components/overview/DetailOverview.ts [hunk 3]
```diff
@@ -17,13 +17,84 @@
...
+  runtime: Runtime = null;
+
+  transactions: TwinTransaction[] = null;
+
+  communicationService: CommunicationService = null;
+
+  async created(): Promise<void> {
+    this.runtime = this.getRuntime();
+    this.communicationService = new CommunicationService(this.runtime);
+    this.twin = this.$store.state.twin;
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/overview/DetailOverview.ts]: would it make sense to define this as computed value?

something like this? 
```
get twin () {
  return this.$store.state.twin;
}
```

File: dapps/digital-twin-detail/src/components/overview/DetailOverview.vue [hunk 4]
```diff
@@ -18,16 +18,34 @@
...
-export default EvanTwinDetailOverviewComponent;
+import DetailOverviewComponent from './DetailOverview';
+
+export default DetailOverviewComponent;
 </script>
 
 <style lang="scss" scoped>
-// @import "EvanTwinDetailOverview.scss";
+.content {
+  margin-left: auto;
```
Comment (karladler) [inline dapps/digital-twin-detail/src/components/overview/DetailOverview.vue]: i think there was some css class adding margin auto... `m-auto'

File: dapps/digital-twin-detail/src/components/overview/DetailOverview.ts (hunk 3 above)
Comment (Tschuck) [reply dapps/digital-twin-detail/src/components/overview/DetailOverview.ts]: Should be okay like this. The twin is set using `$set` within the root component to handle object reactivity of loading flags / update flags.

File: core/evancore.vue.libs/src/component.ts [hunk 5]
```diff
@@ -20,19 +20,47 @@
...
-import VueRouter, { Route } from 'vue-router';
+import VueRouter, { Route, RawLocation } from 'vue-router';
 
 // evan.network imports
 import * as dappBrowser from '@evan.network/ui-dapp-browser';
+import { Runtime } from '@evan.network/api-blockchain-core';
 import { getDomainName } from './utils';
 import { StartedDAppInfo } from './interfaces';
 
+declare module 'vue/types/vue' {
```
Comment (Tschuck) [inline core/evancore.vue.libs/src/component.ts]: Can we move this declaration to our general vue types file?

https://github.com/evannetwork/ui-dapps/blob/feature/CORE-849-digital-twin-general-overview/scripts/dapp/vue-shim.d.ts

File: dapps/digital-twin-detail/src/components/overview/DetailOverview.vue (hunk 4 above)
Comment (datrinh) [reply dapps/digital-twin-detail/src/components/overview/DetailOverview.vue]: I prefer not to use bootstrap classes when it's that easy tbh.

File: dapps/digital-twin-detail/src/components/overview/DetailOverviewTransactions.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+            {{ row.feeInEve }}
+          </td>
+          <td>{{ row.initiator }}</td>
+          <td>{{ row.timestamp || 'TODO' }}</td>
+          <td>
+            <evan-button
+              size="sm"
+              type="icon-secondary"
+              icon="mdi mdi-chevron-right"
+              @click="routeToTransactionExplorer(row.blockHash)"
```
Comment (Tschuck) [inline dapps/digital-twin-detail/src/components/overview/DetailOverviewTransactions.vue]: Would be nice, if we just use the `href` attribute here, so the use can also copy the link directly by right clicking. `evan-button` supports this out of the box :)

Comment (datrinh) [reply dapps/digital-twin-detail/src/components/overview/DetailOverviewTransactions.vue]: Cool, added it.

**Pull request merged**